### PR TITLE
Add download possibility for images

### PR DESCRIPTION
**Pull request by KochTobi**
Requires 
* https://github.com/qbicsoftware/omero-client-lib/pull/10 
* https://github.com/qbicsoftware/omero-client-lib/pull/12

### CODE REVIEW

**Comment by KochTobi** [comment]
@wow-such-code you could call/implement the download link generation in `linkToImageDownload(long imageId)`

**Comment by KochTobi** [comment]
This PR has to wait for a new release of the [omero client lib](https://github.com/qbicsoftware/omero-client-lib).

**Review by Steffengreiner — APPROVED** [review]
LGTM! Nice Work in removing the Json Business from the portlet! However, I'd delay the merge until omero-client-lib provides the omero-tiff download method

File: src/main/java/life/qbic/portal/portlet/OMEROClientPortlet.java [hunk 1]
```diff
@@ -27,43 +25,27 @@
...
 import life.qbic.portal.utils.ConfigurationManagerFactory;
-import omero.model.NamedValue;
+import loci.poi.util.SystemOutLogger;
+import loci.poi.util.TempFile;
+import omero.gateway.model.FileAnnotationData;
 import omero.gateway.model.MapAnnotationData;
+import omero.model.AnnotationAnnotationLink;
+import omero.model.Format;
+import omero.model.ImageI;
+import omero.model.NamedValue;
```
[the remark this answers is not in the document]
Comment (KochTobi) [reply src/main/java/life/qbic/portal/portlet/OMEROClientPortlet.java:46]: I think there is some limit in the IDE on when to group. We need to remove all the omero stuff from the portlet anyways in a future commit.

**Comment by KochTobi** [comment]
@jenniferboedker we could also move your commits to another branch so that the basic functionality of the link is provided and you could replace it in your PR. What do you think?

File: pom.xml [hunk 2]
```diff
@@ -63,7 +63,7 @@
 		<dependency>
 			<groupId>life.qbic</groupId>
 			<artifactId>omero-client-lib</artifactId>
-			<version>1.1.0</version>
+			<version>1.2.0-SNAPSHOT</version>
```
Comment (jenniferboedker) [inline pom.xml:66]: do we still want to increase the version?

Comment (KochTobi) [reply pom.xml:66]: Yes we should increase it to `1.2.0` when we release. `1.1.0` already was released and is the last release there is.

**Comment by KochTobi** [comment]
This PR needs https://github.com/qbicsoftware/omero-lib/pull/12 to be in the `omero-lib`

File: src/main/java/life/qbic/portal/portlet/OMEROClientPortlet.java [hunk 3]
```diff
@@ -417,15 +370,42 @@ private void loadProjects() {
...
+     * @return a vaadin {@link Button} component linking to the download
+     */
+    private Button downloadImage(long imageId) {
+        Button downloadButton = new Button("Download Image");
+        downloadButton.setEnabled(false);
+
+        try{
+            downloadButton.addClickListener(clickEvent -> {
+                try {
+                    String imagePath = omeroClient.downloadOmeTiff(imageId);
```
[the remark this answers is not in the document]
Comment (KochTobi) [reply src/main/java/life/qbic/portal/portlet/OMEROClientPortlet.java:393]: There is a PR open in the lib.

File: src/main/java/life/qbic/portal/portlet/OMEROClientPortlet.java [hunk 4]
```diff
@@ -326,7 +286,6 @@ private void registerListeners() {
                 imageInfos.clear();
                 samples.clear();
                 // load new samples
-                //FIXME please remove crazy HashMap stuff
                 HashMap<Long, HashMap<String, String>> projectSamples = omeroClient
```
Comment (wow-such-code) [inline src/main/java/life/qbic/portal/portlet/OMEROClientPortlet.java:289]: 🤣

**Review by wow-such-code — APPROVED** [review]

**Pull request merged**